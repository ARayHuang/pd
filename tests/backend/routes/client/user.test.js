const request = require('supertest');
const { connect } = require('ljit-db/mongoose');
const config = require('config');
const mongoose = require('mongoose');
const {
	deleteDocuments,
	insertDocuments,
} = require('../../../../server/fixtures/index');
const { app } = require('../../../../server/client');
const LOGIN = {
	MANAGER: {
		PROVIDER: {
			password: '123qwe',
			username: 'provider01',
		},
	},
};
let PROVIDER_MANAGER_COOKIE;

beforeAll(async () => {
	try {
		await connect(config.SERVER.MONGO.URL, { debug: false });
		await deleteDocuments();
		await insertDocuments();

		await request(app)
			.post('/api/v1/login')
			.send(LOGIN.MANAGER.PROVIDER)
			.set('Content-Type', 'application/json')
			.set('Accept', 'application/json')
			.expect(201)
			.then(res => {
				PROVIDER_MANAGER_COOKIE = res.headers['set-cookie'];
			});
	} catch (error) {
		throw error;
	}
});

afterAll(() => {
	return mongoose.connection.close();
});

describe('user', () => {
	describe('get /api/v1/users/id=me', () => {
		test('200 ok', () => {
			return request(app)
				.get('/api/v1/users/id=me')
				.set('Content-Type', 'application/json')
				.set('Accept', 'application/json')
				.set('Cookie', PROVIDER_MANAGER_COOKIE)
				.expect(200)
				.then(res => {
					expect(res.body).toMatchSnapshot();
				});
		});
		test('401 unauthorized', () => {
			return request(app)
				.get('/api/v1/users/id=me')
				.set('Content-Type', 'application/json')
				.set('Accept', 'application/json')
				.expect(401)
				.then(res => {
					expect(res.body).toMatchSnapshot();
				});
		});
	});
});
