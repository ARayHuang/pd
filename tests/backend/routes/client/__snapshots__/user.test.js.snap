// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user get /api/v1/users/id=me 200 ok 1`] = `
Object {
  "channelSettings": Array [],
  "channels": Array [
    Object {
      "id": "5ec5f60b25fbb049dbd231e2",
      "name": "频道一",
    },
  ],
  "departmentType": "provider",
  "displayName": "开单组管理员",
  "id": "5ec4ec469d87c4387d9350ba",
  "profilePictureId": "1",
  "type": "manager",
  "username": "provider01",
}
`;

exports[`user get /api/v1/users/id=me 401 unauthorized 1`] = `
Object {
  "code": "888.010.402",
  "message": "请先登入",
  "type": "AuthenticationError",
}
`;
